toolkit/tools/imagecustomizerapi/README.md

```diff
@@ -0,0 +1,16 @@
+# Mariner Image Customizer API
```

Comment:
I think, we should probably keep all the libraries in `pkg` directory.
The reason, I request this is because, I have proposed that we have a distinction between `main` packages and `lib` packages.

Reference: 
- https://github.com/microsoft/CBL-Mariner/pull/2983

Reply:
I am not sure I am fond of that code layout. Having Go directories with the same name (e.g. `specreader` and `pkg/specreader`) can make writing Go code kind of annoying as it forces you to rename one of the packages in the imports.

I also want to avoid the case where the package name is something overly generic. Hence, why I have `imagecustomizerapi` and not `imagecustomizer/api`.

Reply:
I moved `imagecustomizerlib` into `pkg` but kept `imagecustomizer` and `imagecustomizerapi` here.

Reply:
Only request is that we track any "internal" code via something like we do for the imager
``` makefile
go_internal_files = $(shell find $(TOOLS_DIR)/internal/ -type f -name '*.go')
go_pkg_files = $(shell find $(TOOLS_DIR)/pkg/ -type f -name '*.go')
go_imagegen_files = $(shell find $(TOOLS_DIR)/imagegen/ -type f -name '*.go')
go_scheduler_files = $(shell find $(TOOLS_DIR)/scheduler -type f -name '*.go')
go_common_files = $(go_module_files) $(go_internal_files) $(go_imagegen_files) $(go_pkg_files) $(go_scheduler_files) $(STATUS_FLAGS_DIR)/got_go_deps.flag $(BUILD_DIR)/tools/internal.test_coverage
```

Its not particularly elegant but it makes sure we rebuild all the go tools if any of the shared libraries change.